### PR TITLE
Issue 4: Add /chat/{id}/entry API call

### LINKED ISSUE #4: Send message to LLM and print response in chat
1. It should be possilbe to send request to /chat/{id}/entry with prompt param
2. this message should be saved with role USER
3. this message should be send into LLM chat service
4. The response from LLM should be saved into database with role ASSISTANT

### PR DESCRIPTION
**Pull request by YuliaMittova**
Send message to LLM and print response in chat #4
It should be possible to send a request to /chat/{id}/entry with a prompt param
this message should be saved with role USER
this message should be sent into LLM chat service
The response from LLM should be saved into database with role ASSISTANT
FAIL_TO_PASS: com.example.llmchat.integration.ChatEntryApiIntegrationTest

### CODE REVIEW

**Comment by YuliaMittova** [comment]
@issue-validator validate the issue

**Comment by YuliaMittova** [comment]
@issue-validator validate the issue

**Comment by YuliaMittova** [comment]
@issue-validator validate the issue

**Comment by YuliaMittova** [comment]
@issue-validator validate the issue

**Comment by YuliaMittova** [comment]
@issue-validator validate the issue

**Comment by YuliaMittova** [comment]
@issue-validator validate the issue

**Comment by YuliaMittova** [comment]
@issue-validator validate the issue

**Comment by DmitriyChernyakSetronica** [comment]
@issue-validator validate the issue

**Comment by DmitriyChernyakSetronica** [comment]
green validation is https://github.com/dpaia/infrastructure/actions/runs/20528810131/job/58994500690

**Comment by YuliaMittova** [comment]
@issue-validator validate the issue

**Comment by DmitriyChernyakSetronica** [comment]
@issue-validator validate the issue

**Comment by DmitriyChernyakSetronica** [comment]
ChatEntryApiIntegrationTest doesn't check for saving to the database - doesn't check requirements 2 and 4

File: src/test/kotlin/com/example/llmchat/integration/ChatEntryApiIntegrationTest.kt [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+        val error = JsonPath.read<String>(json, "$.error")
+        assertTrue(error.contains("Bad Request"))
+
+        val message = JsonPath.read<String>(json, "$.message")
+        assertTrue(message.contains("prompt"))
+    }
+
+    @Test
+    fun `when unknown chat id is passed then 404 error is returned`() {
+        val url = "http://localhost:$port/chats/999/entry?prompt=hi"
```
Comment (DmitriyChernyakSetronica) [inline src/test/kotlin/com/example/llmchat/integration/ChatEntryApiIntegrationTest.kt]: url should be /chat/999/entry not /chats/999/entry

File: src/test/kotlin/com/example/llmchat/integration/ChatEntryApiIntegrationTest.kt [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+        assertNotNull(json)
+
+        val chatId = JsonPath.read<Number>(json, "$.chatId").toLong()
+        assertEquals(1, chatId)
+
+        val answer = JsonPath.read<String>(json, "$.answer")
+        assertTrue(answer.contains(ANSWER))
+
+        wiremock.verify(postRequestedFor(urlEqualTo("/api/chat")))
+    }
```
Comment (DmitriyChernyakSetronica) [inline src/test/kotlin/com/example/llmchat/integration/ChatEntryApiIntegrationTest.kt:104]: To check requirements 2 and 4, for example, the agent suggested adding the following block
     
  // Then: Verify messages were saved to database (requirements 2 and 4)
        val chat = chatRepository.findById(chatId).orElseThrow()
        assertTrue(chat.history.size >= 2, "Should have at least USER and ASSISTANT messages")
        
        // Find the messages we just added (last 2 messages)
        val userMessage = chat.history.findLast { it.role == Role.USER && it.content == prompt }
        val assistantMessage = chat.history.findLast { it.role == Role.ASSISTANT && it.content.contains(ANSWER) }
        
        assertNotNull(userMessage, "USER message should be saved to database")
        assertNotNull(assistantMessage, "ASSISTANT message should be saved to database")
        assertEquals(prompt, userMessage.content, "USER message content should match prompt")
        assertTrue(assistantMessage.content.contains(ANSWER), "ASSISTANT message should contain LLM response")
    }

**Comment by DmitriyChernyakSetronica** [comment]
please add test case for "when LLM returns empty response then 502 error is returned"
After this, check the full coverage of integration tests and it is worth removing all similar test units.
I believe this ChatEntryApiIntegrationTest test  alone will be sufficient.

File: src/test/kotlin/com/example/llmchat/integration/ChatEntryApiIntegrationTest.kt (hunk 2 above)
Comment (YuliaMittova) [reply src/test/kotlin/com/example/llmchat/integration/ChatEntryApiIntegrationTest.kt:104]: Fixed in the same commit.

File: src/test/kotlin/com/example/llmchat/integration/ChatEntryApiIntegrationTest.kt (hunk 1 above)
Comment (YuliaMittova) [reply src/test/kotlin/com/example/llmchat/integration/ChatEntryApiIntegrationTest.kt]: Fixed in the same commit

**Comment by YuliaMittova** [comment]
> please add test case for "when LLM returns empty response then 502 error is returned" After this, check the full coverage of integration tests and it is worth removing all similar test units. I believe this ChatEntryApiIntegrationTest test alone will be sufficient.

Added this test case to the integration test. As far as I know as per test pyramid integration tests are not replacing unit tests. But if you want I can delete them.

**Comment by DmitriyChernyakSetronica** [comment]
@issue-validator validate the issue

**Comment by DmitriyChernyakSetronica** [comment]
@issue-validator validate the issue

**Comment by DmitriyChernyakSetronica** [comment]
Verified, accepted

**Comment by kaleev** [comment]
Please extract the addition of test dependencies into a separate base commit.
